### PR TITLE
fix(one-app-runner): have one-app-runner-test print any startup error messages

### PR DESCRIPTION
**Pull request by PixnBits**
## **Description**
Have one-app-runner-test print any startup error messages to the user's terminal and exit quickly when there are errors (resolving #525 except for a configurable timeout) via an [AbortController](https://nodejs.org/api/globals.html#class-abortcontroller).

## **Motivation** 
Currently any issues starting the runner are not reported except for the >3min wait timing out. This is difficult to diagnose, and frustrating that it can take so long.

## **Test** **Conditions**
Despite coverage reporting 100% I did not see an existing test specification file for one-app-runner-test and, possibly embarrassingly, have not added one in this PR. Instead, for now I have copied the two unit files into an existing module's `node_modules` (replacing those placed by `$ npm ci`) and run the `test:a11y` npm script. Rather than failing at the timeout with no `one-app-test.log` file, an error from Docker is displayed on the screen. 
```
$ npm run test:a11y

> my-module@1.2.3 pretest:a11y
> create-runner-env && one-app-runner-test

Waiting for One App to start on port 49281 (timeout of 200000ms)
/private/tmp/my-module/node_modules/@americanexpress/one-app-runner/src/startApp.js:121
      throw new Error(
            ^

Error: Error creating docker network Error: (HTTP code 404) driver not found - could not find an available, non-overlapping IPv4 address pool among the defaults to assign to the network 
    at startApp (/private/tmp/my-module/node_modules/@americanexpress/one-app-runner/src/startApp.js:121:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
failed to start One App, please check any messages above and the file one-app-test.log for more information. Error: one-app-runner exited when it was expected to remain running (exit code 1)
    at ChildProcess.<anonymous> (/private/tmp/my-module/node_modules/@americanexpress/one-app-runner/bin/one-app-runner-test.js:49:9)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1100:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
```
I can now determine I need to prune some networks (`$ docker network prune`).

#### Check boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

#### Check boxes that apply:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

**Review by code-forger — DISMISSED** [review]

**Review by Matthew-Mallimo — DISMISSED** [review]

**Comment by PixnBits** [comment]
Testing the success path today I found that Node.js was kept running by an active handle. Turns out that was the `runnerProcess` STDIO (`runnerProcess.stdout`, `runnerProcess.stderr`). Took a while to identify, then found [`.destroy()`](https://nodejs.org/docs/latest-v18.x/api/stream.html#readabledestroyerror) at the end let Node.js exit, _then_ found that `stdio: 'inherit'` not only manages piping to `process` STDIO but also manages the readable streams for us to let Node.js exit. Thus the seemingly-simple change in 5233b35.

**Comment by PixnBits** [comment]
I had created #539 for the issue of AbortController not being in Node.js 12.x causing test failures, but there's more to it than just changing the versions in the workflows. Should we drop Node.js 12.x support or add a polyfill and conditionally require it? 🤔

**Comment by JAdshead** [comment]
> I had created #539 for the issue of AbortController not being in Node.js 12.x causing test failures, but there's more to it than just changing the versions in the workflows. Should we drop Node.js 12.x support or add a polyfill and conditionally require it? 🤔

I think this should drop node 12 support

**Comment by JAdshead** [comment]
> I had created #539 for the issue of AbortController not being in Node.js 12.x causing test failures, but there's more to it than just changing the versions in the workflows. Should we drop Node.js 12.x support or add a polyfill and conditionally require it? 🤔

It's also missing in 14

**Comment by PixnBits** [comment]
Both 14.x and 12.x removed in #541, no polyfill needed now

**Review by Matthew-Mallimo — APPROVED** [review]

File: packages/one-app-runner/utils/waitForOK.js [hunk 1]
```diff
@@ -1,37 +1,61 @@
...
+      pollingHandle must be defined to build the value of timeoutHandle,
+      pollingHandle depends on this poll function
+    */
+    clearTimeout(timeoutHandle);
+    return resolve(status);
+  }
+
+  const pollingHandle = setInterval(poll, 1000);
+  // the interval will start after the duration between, trying now is desireable
+  setImmediate(poll, 0);
```
Comment (JAdshead) [inline packages/one-app-runner/utils/waitForOK.js:45]: I think the 0 is hangover from setTimeout
```suggestion
  setImmediate(poll);
```

Comment (PixnBits) [reply packages/one-app-runner/utils/waitForOK.js:45]: good catch!!

File: packages/one-app-runner/bin/one-app-runner-test.js [hunk 2]
```diff
@@ -15,29 +15,64 @@
  */
 
 require('dotenv').config();
-const { spawn } = require('child_process');
-const { waitForOK } = require('../utils/waitForOK');
+const { spawn } = require('node:child_process');
```
Comment (bishnubista) [inline packages/one-app-runner/bin/one-app-runner-test.js:18]: What is the difference `node:child_process` vs `child_process?

Comment (JAdshead) [reply packages/one-app-runner/bin/one-app-runner-test.js:18]: the `node` prefix e.g. `node:child_process` is a convention to help identify what packages are natively provided and what need to be installed

**Review by JAdshead — APPROVED** [review]

**Review by Matthew-Mallimo — APPROVED** [review]

**Pull request merged**